### ISSUE TITLE
ctc_batch_cost  "No valid path found." when in batch , but not when not in batch?

### ISSUE THREAD

**Issue by danFromTelAviv** [keras-team/keras#11516]
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I am using ctc_batch_cost  and I keep getting "No valid path found." after a few thousand samples. To find which particular sample in the batch was causing the issue I tried to run the code with batch size of 1. The issue no longer repeats. 

Just to make sure there is no doubt - All of the samples in the first batch in which this error occurs have longer input length compared to label by a large margin and very few repeats so there should definitely be a valid path. I used a seed so that the order of the samples is the same when using a batch size of 1 and a batch size of 32. 

changing the line : 
y_pred = tf.log(tf.transpose(y_pred, perm=[1, 0, 2]) + epsilon()) 
to 
y_pred = tf.log(tf.transpose(y_pred, perm=[1, 0, 2]) + epsilon()) - epsilon()

seems to help - still fails eventually but far less frequently.

**Comment by gabrieldemarmiesse** [comment]
Can you provide a script to reproduce the error. It will speed up the debugging.

**Comment by danFromTelAviv** [comment]
this is the most minimal code I could compile : 


`
```
from keras import Input, Model
from keras.layers import Lambda
import keras.backend as K
from keras.optimizers import Adam


def ctc_lambda_func(args):
    y_pred, labels, input_length, label_length = args
    ''' from TF: Input requirements
    1. sequence_length(b) <= time for all b
    2. max(labels.indices(labels.indices[:, 1] == b, 2)) <= sequence_length(b) for all b.
    '''
    return K.ctc_batch_cost(labels, y_pred, input_length, label_length)

def ctc(y_true, y_pred):
    return y_pred

def net(num_features_in):
    input_layer = Input(shape=(None, num_features_in), name="the_input")
    output= Lambda(lambda x:x)(input_layer)
    labels = Input(name='the_labels', shape=[None, ], dtype='int32')
    input_length = Input(name='input_length', shape=[1], dtype='int32')
    label_length = Input(name='label_length', shape=[1], dtype='int32')
    loss_out = Lambda(ctc_lambda_func, output_shape=(1,), name='ctc')([output,
                                                                       labels,
                                                                       input_length,
                                                                       label_length])
    model = Model(inputs=[input_layer, labels, input_length, label_length], outputs=loss_out)
    opt = Adam()
    model.compile(optimizer=opt, loss=ctc)
    return model
```
`
